src/modal/modal-container.component.ts

```diff
@@ -38,6 +38,7 @@ export class ModalContainerComponent implements OnInit, OnDestroy {
   isAnimated: boolean;
   bsModalService: BsModalService;
   private isModalHiding = false;
+  private clickStartedInside = true;
```

Comment:
The default value should be `false` because if there is no mousedown (started outside) the constraint is not checked if it started inside but value would be true.